### PR TITLE
Aerogear 2668

### PR DESCRIPTION
**Pull request by StephenCoady**
## Motivation

JIRA: https://issues.jboss.org/browse/AEROGEAR-2668
MAIL: groups.google.com/forum/#!topic/aerogear/1hxhEPBASVY

## Description

Added two new methods, one to return an array of service configs based on type and another to return the first instance found.

## TODO
- [x] Add new find first service config by type method
- [x] Add new find service config by type - returning an array
- [x] Add tests

### CODE REVIEW

File: Core/Core/MobileCore.cs [hunk 1]
```diff
@@ -213,9 +212,29 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
             throw new ServiceModuleInstanceNotFoundException(String.Format("No instance has been registered for interface {0}", serviceClass.Name));
         }
 
-        public ServiceConfiguration GetServiceConfiguration(String type)
+        public ServiceConfiguration[] GetServiceConfigurationByType(String type)
```
Comment (wei-lee) [inline Core/Core/MobileCore.cs]: can we add some comments to the public methods

File: Core/Core/MobileCore.cs [hunk 2]
```diff
@@ -213,9 +212,29 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
...
+                    listOfConfigs.Add(item.Value);
+                }
+            }
+            ServiceConfiguration[] arrayOfConfigs = listOfConfigs.ToArray();
+            return arrayOfConfigs;
+        }
+
+        public ServiceConfiguration GetFirstServiceConfigurationByType(String type)
         {
             return servicesConfig.ContainsKey(type) ? servicesConfig[type] : null;
```
Comment (wei-lee) [inline Core/Core/MobileCore.cs]: I don't think the implementation here is right? This should call `GetServiceConfigurationByType` and return the first one in the array?

**Comment by StephenCoady** [comment]
@wei-lee, thanks - made those changes

File: Core/Core/MobileCore.cs [hunk 3]
```diff
@@ -213,10 +212,44 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
...
-        public ServiceConfiguration GetServiceConfiguration(String type)
+        /// <summary>
+        /// Returns array of ServiceConfiguration, filtered by type
+        /// </summary>
+        /// <param name="type">type field of the configuration</param>
+        /// <returns>array of ServiceConfiguration</returns>
+        public ServiceConfiguration[] GetServiceConfigurationByType(String type)
+        {
+            List<ServiceConfiguration> listOfConfigs = new List<ServiceConfiguration>();
+            foreach (var item in servicesConfig)
```
Comment (josemigallas) [inline Core/Core/MobileCore.cs]: Why not using Linq?

File: Core/Core/MobileCore.cs [hunk 4]
```diff
@@ -213,10 +212,44 @@ private T GetInstance<T>(Type serviceClass, ServiceConfiguration serviceConfigur
...
+                {
+                    listOfConfigs.Add(item.Value);
+                }
+            }
+            ServiceConfiguration[] arrayOfConfigs = listOfConfigs.ToArray();
+            return arrayOfConfigs;
+        }
+
+        /// <summary>
+        /// Returns the first instance of a ServiceConfiguration based on the type key
```
Comment (josemigallas) [inline Core/Core/MobileCore.cs]: This is a rather picky observation but what does exactly define "first" here?. There is no order in `servicesConfig`, right? Is it the same as saying "first occurrence"?

**Comment by josemigallas** [comment]
For what it's worth, LGTM

**Pull request merged**